lib/ts/superTokens.tsx

```diff
@@ -297,35 +297,26 @@ export default class SuperTokens {
         throw new Error("Should never come here: unexpected redirection context");
     }
 
-    redirectToAuthWithRedirectToPath = (show?: "signin" | "signup", history?: any, queryParams?: any) => {
-        const redirectToPath = getCurrentNormalisedUrlPath().getAsStringDangerous();
-        if (queryParams === undefined) {
-            queryParams = {};
+    redirectToAuth = async (options: {
+        show?: "signin" | "signup";
+        history?: any;
+        queryParams?: any;
+        redirectBack?: boolean;
+    }) => {
+        const queryParams = options.queryParams === undefined ? {} : options.queryParams;
+        if (options.show !== undefined) {
+            queryParams.show = options.show;
         }
-        queryParams = {
-            ...queryParams,
-            redirectToPath,
-        };
-        return this.redirectToAuthWithoutRedirectToPath(show, history, queryParams);
-    };
-
-    redirectToAuthWithoutRedirectToPath = async (show?: "signin" | "signup", history?: any, queryParams?: any) => {
-        if (queryParams === undefined) {
-            queryParams = {};
-        }
-        if (show !== undefined) {
-            queryParams = {
-                ...queryParams,
-                show,
-            };
+        if (options.redirectBack === true) {
+            queryParams.redirectToPath = getCurrentNormalisedUrlPath().getAsStringDangerous();
```

Comment:
Default value of redirectBack should be true. By doing this check this way, the default value is being set to false.

Reply:
@porcellus missed this?

Reply:
Since `redirectBack` is now required, I don't think this applies anymore, but I'm happy to make it an implicit check instead.